cmd/agent/main.go

```diff
@@ -13,7 +14,31 @@ import (
 	"github.com/spf13/cobra"
 )
 
-const lockFile = "/var/run/xk6-disruptor"
+const lockFile = "xk6-disruptor"
+
+// returns the path to the lock file
+func getLockPath() string {
+	// get runtime directory for user
+	lockDir := os.Getenv("XDG_RUNTIME_DIR")
+	if lockDir == "" {
+		lockDir = fmt.Sprintf("/run/user/%d", os.Geteuid())
+	}
```

Comment:
I think we might need to refine this logic a bit. For example, it wouldn't be strange for minimal distro images to not include `/run/user/$id`, or even `/run/user` altogether, which may cause this step to fail.

Perhaps the most reliable way after all would be to just `os.Mkdir(path.Join(os.TempDir(), "xk6-disruptor"))` and use that. I think it's reasonable to believe that, contrary to `/run` or `/var/run`, all users can create files in `/tmp`.


```
17:10:24   ~ $> docker run --rm -i alpine ls -la /run
total 9
drwxr-xr-x    2 root     root             2 May  9 18:39 .
drwxr-xr-x   19 root     root            20 May 29 15:10 ..

roobre@Archiroo
17:10:50   ~ $> docker run --rm -i ubuntu ls -la /run
Status: Downloaded newer image for ubuntu:latest
total 11
drwxr-xr-x  5 root root  5 Apr 25 14:06 .
drwxr-xr-x 17 root root 24 May 29 15:11 ..
drwxrwxrwt  2 root root  2 Apr 25 14:03 lock
drwxr-xr-x  2 root root  2 Apr 25 14:03 mount
drwxr-xr-x  2 root root  3 Apr 25 14:06 systemd

```

Reply:
The main use case of the agent is to run in the images we provide, on which we should ensure this requirement is satisfied. I would like to keep this as the main focus. I don't think we should introduce more requirements to make this agent a generic command line. 

The second use case is the debugging of the agent, which I expect to happen in a "regular" distribution. In this scenario, I acknowledged that using `XDG_RUNTIME_DIR` was the best option so I introduced this change.

As for other corner uses cases such as the one you mention, there are other requirements for running the agent that maybe are not satisfied in those distros (e.g. `iptables`).

I prefer to keep it simple to its intended use case. If at some point someone wants to use it in another context, then it would be the moment to reconsider these restrictions)

Reply:
That's a good point, I lost focus of the approach for a sec and forgot that we run this on a dedicated image.

That being said, I'm happy to see `XDG_RUNTIME_DIR` being honored, but what I'm not fully sold on is on defaulting on a directory that may not exist. From what I've seen, if `XDG_RUNTIME_DIR` is set, then `/run/user` (most likely) exists, but if `XDG_RUNTIME_DIR` is not set, then (most likely) `/run/user` will not exist either. That makes me think that the fallback logic is effectively noop.

It makes more sense to me to honor the stander if possible (when `XDG_RUNTIME_DIR` is set), and if it isn't, pick a safer default that has more chances of succeeding, like `os.Tempdir()`. If we choose to not care about non-controlled environments, which I agree it's a good approach, we could exit with an error if `XDG_RUNTIME_DIR` is not set.

Reply:
>If we choose to not care about non-controlled environments, which I agree it's a good approach, we could exit with an error if XDG_RUNTIME_DIR is not set.

What you propose makes sense. However, the problem is that the image we are using now, `alpine:3.15` does not honor this standard. And this makes sense because it is not intended to be a desktop environment. We could introduce this variable in the image, but that seems artificial to me (as the image won't honor the other environment variables defined in the standard). 

So, it will end up that the main use case uses the fall-back location. Not sure if this makes sense.

This takes me to my original point that we are changing things to cover use cases that are out of scope. 

I prefer then to honor the variable if present and if not, default to a known location in the main target (the alpine image)